test/e2e/storage/vsphere/vsphere_utils.go

```diff
@@ -443,27 +410,321 @@ func getvSphereVolumePathFromClaim(client clientset.Interface, namespace string,
 	return pv.Spec.VsphereVolume.VolumePath
 }
 
-func addNodesToVCP(vsp *vsphere.VSphere, c clientset.Interface) error {
-	nodes, err := c.CoreV1().Nodes().List(metav1.ListOptions{})
+// Get canonical volume path for volume Path.
+// Example1: The canonical path for volume path - [vsanDatastore] kubevols/volume.vmdk will be [vsanDatastore] 25d8b159-948c-4b73-e499-02001ad1b044/volume.vmdk
+// Example2: The canonical path for volume path - [vsanDatastore] 25d8b159-948c-4b73-e499-02001ad1b044/volume.vmdk will be same as volume Path.
+func getCanonicalVolumePath(ctx context.Context, dc *object.Datacenter, volumePath string) (string, error) {
+	var folderID string
+	canonicalVolumePath := volumePath
+	dsPathObj, err := getDatastorePathObjFromVMDiskPath(volumePath)
 	if err != nil {
-		return err
+		return "", err
 	}
-	for _, node := range nodes.Items {
-		vsp.NodeAdded(&node)
+	dsPath := strings.Split(strings.TrimSpace(dsPathObj.Path), "/")
+	if len(dsPath) <= 1 {
+		return canonicalVolumePath, nil
 	}
-	return nil
+	datastore := dsPathObj.Datastore
+	dsFolder := dsPath[0]
+	// Get the datastore folder ID if datastore or folder doesn't exist in datastoreFolderIDMap
+	if !isValidUUID(dsFolder) {
+		dummyDiskVolPath := "[" + datastore + "] " + dsFolder + "/" + DummyDiskName
+		// Querying a non-existent dummy disk on the datastore folder.
+		// It would fail and return an folder ID in the error message.
+		_, err := getVirtualDiskPage83Data(ctx, dc, dummyDiskVolPath)
+		if err != nil {
+			re := regexp.MustCompile("File (.*?) was not found")
+			match := re.FindStringSubmatch(err.Error())
+			canonicalVolumePath = match[1]
+		}
+	}
+	diskPath := getPathFromVMDiskPath(canonicalVolumePath)
+	if diskPath == "" {
+		return "", fmt.Errorf("Failed to parse canonicalVolumePath: %s in getcanonicalVolumePath method", canonicalVolumePath)
+	}
+	folderID = strings.Split(strings.TrimSpace(diskPath), "/")[0]
+	canonicalVolumePath = strings.Replace(volumePath, dsFolder, folderID, 1)
+	return canonicalVolumePath, nil
+}
+
+// GetPathFromVMDiskPath retrieves the path from VM Disk Path.
```

Comment:
No a public method, so will this capitalization give any problem?

Reply:
Fixed capitalization in func documentation